app/Platform/Commands/SyncGameSets.php

```diff
@@ -4,35 +4,120 @@
+                );
+            }
+
+            $progressBar->advance();
+        }
+        $progressBar->finish();
+
+        $gameAltsCount = GameAlternative::count();
+        $this->info("\nPopulating {$gameAltsCount} game_set_games and game_set_links...");
 
-    // protected function preProcessEntity(object $origin, array $transformed): array
-    // {
-    // }
+        $progressBar = $this->output->createProgressBar($gameAltsCount);
 
-    // protected function postProcessEntity(object $origin, object $transformed): void
-    // {
-    // }
+        foreach (GameAlternative::cursor() as $gameAlt) {
+            $parentGame = Game::find($gameAlt->gameID);
+            $childGame = Game::find($gameAlt->gameIDAlt);
+
+            // Determine if a swap is needed to ensure the hub is always the parent.
+            if (
+                ($parentGame->ConsoleID !== System::Hubs && $childGame->ConsoleID === System::Hubs)
+                || ($gameAlt->gameID > $gameAlt->gameIDAlt)
+            ) {
+                // Swap to ensure the hub is the parent, or ensure consistent ordering.
+                $temp = $parentGame;
+                $parentGame = $childGame;
+                $childGame = $temp;
+            }
+
+            // Ensure the hub is always the parent in the relationship.
+            if ($childGame->ConsoleID === System::Hubs && $parentGame->ConsoleID !== System::Hubs) {
+                // Swap the parent and child so that the hub is always the parent.
+                $temp = $parentGame;
+                $parentGame = $childGame;
+                $childGame = $temp;
+            }
+
+            $parentGameSet = GameSet::firstWhere('game_id', $parentGame->id);
+            $childGameSet = GameSet::firstWhere('game_id', $childGame->id);
+
+            if ($parentGame->ConsoleID === System::Hubs && $childGame->ConsoleID === System::Hubs) {
+                GameSetLink::upsert([
+                    'parent_game_set_id' => $parentGameSet->id,
+                    'child_game_set_id' => $childGameSet->id,
+                    'created_at' => $childGame->Created,
+                    'updated_at' => $childGame->Updated,
+                ], uniqueBy: ['parent_game_set_id', 'child_game_set_id']);
+            } else {
+                GameSetGame::upsert([
+                    'game_set_id' => $parentGameSet->id,
+                    'game_id' => $childGame->id,
+                    'created_at' => $childGame->Created,
+                    'updated_at' => $childGame->Updated,
+                ], uniqueBy: ['game_set_id', 'game_id']);
+
+                // Ensure bi-directionality for non-hub games.
+                GameSetGame::upsert([
+                    'game_set_id' => $childGameSet->id,
+                    'game_id' => $parentGame->id,
+                    'created_at' => $parentGame->Created,
+                    'updated_at' => $parentGame->Updated,
+                ], uniqueBy: ['game_set_id', 'game_id']);
```

Comment:
hubs are somehow still getting added to the Similar Games game set:
```
select g.ConsoleID, g.Title from game_set_games gsg join GameData g on g.ID=gsg.game_id where gsg.game_set_id IN (select id from game_sets where game_id=1471);
```

Reply:
I've made some changes in latest that seem to remediate this issue.